SimTKcommon/tests/TestParallelExecutor.cpp

```diff
@@ -91,10 +91,15 @@ void testSingleThreadedExecution() {
         ASSERT(flags[j] == (j < numFlags-10 ? 1 : 0));
 }
 
+void testResizeThreads() {
+  ParallelExecutor executor(2);
+  
```

Comment:
This needs to verify that the number of threads actually changed and throw an exception if not. That way if someone breaks it in the future we will find out right away. You can at least ask ParallelExecutor for the current number of threads or something like that.